en/08_Changelogs/6.1.0.md

```diff
@@ -99,6 +100,27 @@ If you are using the [`DataObjectSchema::databaseIndexes()`](api:SilverStripe\OR
 
 See the [indexes documentation](/developer_guides/model/indexes/) for more details.
 
+### Generated column support {#generated-columns}
+
+Generated columns are database columns where the value is generated inside the database, rather than being set by a user. They're usually based on other columns in the database and can be either generated when the record is updated and stored, or generated when requested in which case they aren't stored.
+
+Some good use cases for generated columns include:
+
+- sorting in a gridfield on a complex summary field: It's common to use a getter method to get some value derived from your database fields (e.g. a discounted price), and use that in `summary_fields`. With a generated column, you can remove the getter method and the [`GridFieldSortableHeader`](api:SilverStripe\Forms\GridField\GridFieldSortableHeader) component will be able to sort using the column.
+- using functional indexes: generated columns can be included in indexes, allowing you to sort or filter by complex expressions in an efficient way.
+- data integrity: unlike generating values inside an `onBeforeWrite()` extension hook, generated column values will be correct even if you update the record with raw SQL.
+- reduce repetition: instead of using a complex expression in multiple different places, you can just give the expression a name with a generated column.
```

Comment:
I don't understand what this means?

Normally you'd just have a getMyValue() method on a model? i.e. logic in a single place?

Reply:
If you have multiple SQL queries that use the same complex expression, you can replace those complex expressions with a single named generated column.

Reply:
- I have some record.
- I want to filter by an expression in a template, so I add a controller method that does a custom SQL query that uses that expression. Even something simple like `CONCAT("FieldOne", "FieldTwo")`.
- In another area of the code that's completely separate from the above controller, I am using the same `DataObject` model. I want to filter or sort by that same expression.

Instead of having `CONCAT("FieldOne", "FieldTwo")` in both places, I can just add a generated column named `FieldConcat` and use that field name instead.